### ISSUE TITLE
In controlled mode, close button with step incrementation does not show beacon of the next step

### ISSUE THREAD

**Issue by SalvadorDalia** [gilbarbara/react-joyride#1078]
## 🐛 Bug Report

- Using placement "center" in controlled mode and closing tutorial with "x" button, reveals the next step with popup, instead of just showing the beacon. 
- The value of the property "disableBeacon" for the first step doesn't change the outcome. 
- If the first step has placement "top" and target "body", closing tutorial with "x" button behaves as expected - the next step shows only beacon. However this is not a good hotfix, because it moves the popup to the bottom of the page.

## To Reproduce

Based on the basic example on CodeSandbox ([src/Basic/index.tsx](https://codesandbox.io/p/devbox/cranky-tristan-n3wwsd?file=%2Fsrc%2FBasic%2Findex.tsx&workspaceId=036b4ae2-9e21-4f8e-ba6b-e5dee251b7fb))

1. Use controlled mode
2. Add steps with following properties: 
```typescript
[
  {
    // ...
    placement: 'center',
    target: 'body',
    disableBeacon: true,
  },
  {
     // ...
    target: '.star-burst',
    disableBeacon: false,
  },
  {
    // ...
    placement: 'bottom',
    target: '.demo__projects h2',
    disableBeacon: false,
  },
]
 ```
3. In the preview click "Start"
4. In the first step click "x" button - the issue occurs, next step is opened instead of showing beacon
5. In the second step click "x" button - correct behaviour, popup disappears, the third step shows only beacon

## Expected behavior

When first step is closed by "x" button, the second step shows beacon instead of popup

## Link to repl or repo (highly encouraged)

https://codesandbox.io/p/devbox/cranky-tristan-n3wwsd?workspaceId=036b4ae2-9e21-4f8e-ba6b-e5dee251b7fb

## Run `npx envinfo --system --binaries --npmPackages react-joyride`

Paste the results here:

```bash
System:
    OS: Linux 6.1 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (2) x64 AMD EPYC
    Memory: 1.53 GB / 4.01 GB
    Container: Yes
    Shell: Unknown
  Binaries:
    Node: 20.16.0 - /usr/local/bin/node
    Yarn: 1.22.22 - /usr/local/bin/yarn
    npm: 10.8.1 - /usr/local/bin/npm
    pnpm: 9.7.0 - /usr/local/bin/pnpm
  npmPackages:
    react-joyride: ^2.9.0 => 2.9.0
```

**Comment by gilbarbara** [comment]
Hey @SalvadorDalia 
Fixed in 2.9.1

**Comment by SalvadorDalia** [comment]
Hi, @gilbarbara 
Thanks for addressing this so promptly. Your quick fix was exactly what was needed.

I updated the package in my example code and I saw that in the new version, there is a problem with overlay on the first step. The overlay is not showing until the second step is shown. Should I open a new issue for this?

**Comment by gilbarbara** [comment]
Fixed in v2.9.2
